lib/staging/slac/fsm/evse/src/states/others.cpp

```diff
@@ -76,6 +83,57 @@ bool ResetState::handle_slac_message(slac::messages::HomeplugMessage& message) {
     }
 }
 
+void ResetChipState::enter() {
+    ctx.log_info("Entered HW Chip Reset state");
+}
+
+FSMSimpleState::HandleEventReturnType ResetChipState::handle_event(AllocatorType& sa, Event ev) {
+    if (ev == Event::SLAC_MESSAGE) {
+        if (handle_slac_message(ctx.slac_message_payload)) {
+            return sa.create_simple<IdleState>(ctx);
+        } else {
+            return sa.PASS_ON;
+        }
+    } else {
+        return sa.PASS_ON;
+    }
+}
+
+FSMSimpleState::CallbackReturnType ResetChipState::callback() {
+    const auto& cfg = ctx.slac_config;
+    if (not reset_delay_done) {
+        reset_delay_done = true;
+        return cfg.chip_reset_delay_ms;
+    } else if (chip_reset_has_been_sent == false) {
+        slac::messages::cm_reset_device_req set_reset_req;
+
+        ctx.log_info("Resetting HW Chip using RS_DEV.REQ");
+
+        // RS_DEV.REQ is on protocol version 0
+        ctx.send_slac_message(cfg.plc_peer_mac, set_reset_req, 0);
+
+        chip_reset_has_been_sent = true;
+
+        return cfg.chip_reset_timeout_ms;
+    } else {
+        ctx.log_info("RS_DEV.REQ timeout, no response received - failed to reset the chip");
+        return {};
+    }
```

Comment:
Instead of having two bools, I would introduce an `enum class` for this small sub-state machine like
```
enum class ChipResetState {
  DELAY,
  SEND_RESET,
  DONE,
}
```
and have only a state variable of its type.  Checking the bools could also be consistent (either `not` or `== false`).

Reply:
done